src/cocotb/regression.py

```diff
@@ -134,28 +176,39 @@ async def func(*args, **kwargs):
         self.stage = stage
         self.name = self.func.__qualname__ if name is None else name
         self.module = self.func.__module__ if module is None else module
+        self.doc = self.func.__doc__ if doc is None else doc
+        if self.doc is not None:
+            self.doc = inspect.cleandoc(self.doc)
         self.fullname = f"{self.module}.{self.name}"
-        self.included: Optional[bool] = None
```

Comment:
`Test.included` was used to explicitly run a test decorated with `skip=True` by including it in `TESTCASE`.

`test_skipped` is failing now:
https://github.com/cocotb/cocotb/actions/runs/7936272852/job/21671195158?pr=3717#step:23:3660

Reply:
Addressed in the most recent commit (which is a kinda large refactor of RegressionManager). When `TESTCASE` is used, the `TESTCASE` mode is set on the Regression Manager, which causes skipped tests to run.